src/plugins/validation/2and3/semantic-validators/info.js

```diff
@@ -0,0 +1,33 @@
+// Assertation 1:
+// check if info exists
+
+// Assertation 2:
+// making sure that the required version and title are defined properly
+module.exports.validate = function({ jsSpec }) {
+  const errors = [];
+  const warnings = [];
+  const no_input = !jsSpec.info;
+  let mistyped_title_or_version = false;
+  if (no_input) {
+    errors.push({
+      path: ['info'],
+      message: 'Missing info object for api defintion'
+    });
+  }
+  //Assertion 2
+  if (
+    jsSpec.info &&
+    (typeof jsSpec.info.title != 'string' ||
+      typeof jsSpec.info.version != 'string')
```

Comment:
I have a few comments about this validation:

In the first check, you are making sure there is an info section that is an object. So, you can wrap the next validations in an `else` block or something and that way you don't have to check for `jsSpec.info` anymore.

To make this more readable, it's good to use variables with names reflecting what you are actually testing for. Also would be good to make sure the strings aren't empty. So something like
```js
const title = jsSpec.info.title;
const hasTitle = typeof title === 'string' && title.toString().trim().length > 0;

if (!hasTitle) { // ...
```

Also, I think you should split these validations up so they each have their own check and message.